app/src/main/java/com/eveningoutpost/dexdrip/Services/WixelReader.java

```diff
@@ -332,7 +408,13 @@ static Long timeForNextRead() {
             // A new packet should arrive but chance is we have missed it...
             return 30000L;
         }
-        return (DEXCOM_PERIOD - gapTime) + 2000;
+
```

Comment:
Well it depends how much BW and power matter to you compared to having the data faster. I like to have the data on my tablet before dexcom receiver plays the notification tone. Since I'm not going to have an http client, I don't really mind.
I think that if you want a simple solution that will work faster when there are wifi receivers, you can simply count the number of wifi receivers (when parsing the data in any case) and if there are both wifi and http, act faster. If there is *only* http client use 10 seconds.
As I said, I don't really mind, so if you want to know more, let me know.